sqlglot/parser.py

```diff
@@ -1558,6 +1558,27 @@ def _parse_create_table_index(self) -> t.Optional[exp.Expression]:
             amp=amp,
         )
 
+    def _parse_table_parts(self, schema: bool = False) -> exp.Expression:
```

Comment:
I factored this out so that I could override it in TSQL and parse the `FOR SYSTEM_TIME` before the alias.